core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -73,11 +74,20 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
 
   private val indexMetadataStore : Option[IndexMetadataStore] = {
     downsampleConfig.indexMetastoreImplementation match {
-      case IndexMetastoreImplementation.NoImp => None
-      case IndexMetastoreImplementation.File => Some(
-        new FileCheckpointedIndexMetadataStore(downsampleConfig.indexLocation.get)
-      )
-      case IndexMetastoreImplementation.Ephemeral => Some(new EphemeralIndexMetadataStore())
+      case IndexMetastoreImplementation.NoImp       => None
+      case IndexMetastoreImplementation.File        =>
+        val expectedVersion = System.getenv(FileSystemBasedIndexMetadataStore.expectedGenerationEnv)
+        val expectedVersionOption = if (expectedVersion != null) {
+            logger.info("Expected version for the index from env is {}", expectedVersion)
+            Try(Integer.parseInt(expectedVersion)) match {
+              case Success(parsedVersion)   => Some(parsedVersion)
+              case Failure(e)               => logger.warn("Failed while parsing index generation", e)
+                                               None
```

Comment:
NIT: Wondering if we should move this to a separate function and add a UT for this ?

Quick Question: what is the difference between the expectedVersion vs expectedVersionOption ?

Reply:
Good point, let me add one